pvlib/spectrum/mismatch.py

```diff
@@ -113,32 +117,132 @@ def get_am15g(wavelength=None):
+    tilted surface, optionally interpolated to the specified wavelength(s).
 
-        am15g = pd.Series(data=interpolator(wavelength), index=wavelength)
+    Parameters
+    ----------
+    wavelengths : numeric, optional
+        Wavelengths at which the spectrum is interpolated.
+        If not provided, the 2002 wavelengths of the standard are returned.
+        :math:`nm`.
+        Values outside of the range :math:`[280, 4000]` are filled with zeroes.
+
+    reference : str, default "ASTM G173-03"
+        The reference standard spectrum to use. Only the reference spectrum
+        ``"ASTM G173-03"`` is available at the moment.
+
+    Returns
+    -------
+    standard_spectrum : pandas.DataFrame
+        The standard spectrum by ``wavelength [nm]``, in
+        :math:`\frac{W}{m^2 nm}`. Column names are ``extraterrestrial``,
+        ``direct`` and ``global``.
 
-    am15g.index.name = 'wavelength'
-    am15g.name = 'am15g'
+    Notes
+    -----
+    If ``wavelength`` is specified, linear interpolation is used.
+
+    If the values in ``wavelength`` are too widely spaced, the integral of each
+    spectrum may deviate from its standard value.
+    For global spectra, it is about :math:`1000.37 W/m^2`.
+
+    The values of the ASTM G173-03 provided with pvlib-python are copied from
+    an Excel file distributed by NREL, which is found here:
+    https://www.nrel.gov/grid/solar-resource/assets/data/astmg173.xls
+
+    More information about reference spectra is found at [2]_.
+
+    Examples
+    --------
+    >>> from pvlib import spectrum
+    >>> am15 = spectrum.get_standard_spectrum()
+    >>> am15_extraterrestrial, am15_global, am15_direct = \
+    >>>     am15['extraterrestrial'], am15['global'], am15['direct']
+    >>> print(am15.head())
+                extraterrestrial        global        direct
+    wavelength
+    280.0                  0.082  4.730900e-23  2.536100e-26
+    280.5                  0.099  1.230700e-21  1.091700e-24
+    281.0                  0.150  5.689500e-21  6.125300e-24
+    281.5                  0.212  1.566200e-19  2.747900e-22
+    282.0                  0.267  1.194600e-18  2.834600e-21
+
+    >>> am15 = spectrum.get_standard_spectrum([300, 500, 800, 1100])
+    >>> print(am15)
+                extraterrestrial   global    direct
+    wavelength
+    300                  0.45794  0.00102  0.000456
+    500                  1.91600  1.54510  1.339100
+    800                  1.12480  1.07250  0.988590
+    1100                 0.60000  0.48577  0.461130
+
+    References
+    ----------
+    .. [1] ASTM "G173-03 Standard Tables for Reference Solar Spectral
+       Irradiances: Direct Normal and Hemispherical on 37° Tilted Surface."
+    .. [2] “Reference Air Mass 1.5 Spectra,” www.nrel.gov.
+       https://www.nrel.gov/grid/solar-resource/spectra-am1.5.html
+    """  # Contributed by Echedey Luis, inspired by Anton Driesse (get_am15g)
```

Comment:
:)

Reply:
I mean, this function is just a copy-paste of yours with some little changes and substituting the interpolator due to the scipy deprecation. Even half the docs and behaviour is the same. 😉